openlibrary/core/wikidata.py

```diff
@@ -0,0 +1,147 @@
+"""
+The purpose of this file is to:
+1. Interact with the Wikidata API
+2. Store the results
+3. Make the results easy to access from other files
+"""
+
+import requests
+from dataclasses import dataclass
+from openlibrary.core.helpers import days_since
+
+from datetime import datetime
+import json
+from openlibrary.core import db
+
+WIKIDATA_API_URL = 'https://www.wikidata.org/w/rest.php/wikibase/v0/entities/items/'
+WIKIDATA_CACHE_TTL_DAYS = 30
+
+
+@dataclass
+class WikidataEntity:
+    """
+    This is the model of the api response from WikiData plus the updated field
+    https://www.wikidata.org/wiki/Wikidata:REST_API
+    """
+
+    id: str
+    type: str
+    labels: dict
+    descriptions: dict
+    aliases: dict
+    statements: dict
+    sitelinks: dict
+    updated: datetime  # This is when we fetched the data, not when the entity was changed in Wikidata
+
+    def description(self, language: str = 'en') -> str | None:
+        """If a description isn't available in the requested language default to English"""
+        return self.descriptions.get(language) or self.descriptions.get('en')
+
+    @classmethod
+    def from_dict(cls, response: dict, updated: datetime):
+        return cls(
+            id=response['id'],
+            type=response['type'],
+            labels=response['labels'],
+            descriptions=response['descriptions'],
+            aliases=response['aliases'],
+            statements=response['statements'],
+            sitelinks=response['sitelinks'],
+            updated=updated,
+        )
+
+    def as_api_response_str(self) -> str:
```

Comment:
We can rename this to:

```suggestion
    def to_json(self) -> str:
```

or we might not even need a method for this, and just use `json.dumps(wikidata_entity.asdict())`. It'll include the `_updated` field in postgres, but that's fine.

Reply:
If we try to do `json.dumps(asdict(entity))` then we get `Object of type datetime is not JSON serializable`.

However, I did rename it to something that seems clearer: `to_wikidata_api_json_format`